exercises/hangman/hangman_test.py

```diff
@@ -0,0 +1,95 @@
+import unittest
+
+from hangman import Hangman
+
+
+# Tests adapted from csharp//hangman/HangmanTest.cs
+
+class HangmanTests(unittest.TestCase):
+    def test_initially_9_failures_are_allowed(self):
+        game = Hangman('foo')
+        self.assertEqual(game.get_status(), "ongoing")
+        self.assertEqual(game.remainingGuesses, 9)
+
+    def test_initially_no_letters_are_guessed(self):
+        game = Hangman('foo')
+
+        self.assertEqual(game.get_masked_word(), '___')
+
+    def test_after_10_failures_the_game_is_over(self):
+        game = Hangman('foo')
+
+        for i in range(10):
+            game.guess('x')
+
+        self.assertEqual(game.get_status(), 'lose')
+        try:
+            game.guess('x')
+        except Exception as e:
```

Comment:
I'm a little surprised `flake8` doesn't complain about this, but it's not considered best practice to `except Exception`; it is better to be more specific about which exception you're expecting. If you really have to, use `except BaseException` as `Exception` is only a subset of `BaseException`.

All that aside, `unittest` has a method for expected exceptions, and this track uses a variant of that:
```
def do_the_thing():
    raise ValueError('bad stuff happened!')

# ...
	def test_should_raise_exception(self):
		with self.assertRaisesWithMessage(ValueError):
            do_the_thing()

    # Utility functions
    def setUp(self):
        try:
            self.assertRaisesRegex
        except AttributeError:
            self.assertRaisesRegex = self.assertRaisesRegexp

    def assertRaisesWithMessage(self, exception):
        return self.assertRaisesRegex(exception, r".+")
```

A couple of things in there:
- `assertRaisesRegex` is `assertRaisesRegexp` in Python 2, so we check for both
- RE: [track README](https://github.com/exercism/python/blob/054ba4c3ba4fd501129f203a22d7431cb55e465b/README.md#conventions), we don't check for specific error messages here, only that one was provided.